bff-service/index.js

```diff
@@ -0,0 +1,66 @@
+const express = require('express');
+const dotenv = require('dotenv');
+const axios = require('axios').default;
+const cache = require('./cache');
+
+dotenv.config();
+const port = process.env.PORT || 3001;
+
+const app = express();
+app.use(express.json());
+
+app.all('/*', cache(120, 'products'), (req, res) => {
+  console.log('originalUrl', req.originalUrl); // /products/main?res=all
+  console.log('method', req.method); // POST, GET
+  console.log('body', req.body); // { name: 'product-1', count: 12 }
+
+  const { originalUrl, method, body } = req;
+
+  const delimiters = /\/|\?/; // delimiters ['/', '?']
+  const recipient = originalUrl.split(delimiters)[1];
+  console.log('recipient', recipient);
+  const reminder = originalUrl.slice(recipient.length + 2); // '/recipient/'
+  console.log('reminder', reminder);
+
+  const recipientUrl = process.env[recipient];
+  console.log('recipientUrl', recipientUrl);
+
+  if (recipientUrl) {
+    const axiosConfig = {
+      method,
+      url: `${recipientUrl}/${reminder}`,
```

Comment:
In case your origial url is smth like this `/cart/internal-path` then everything working fine.
However for url that contains only query parameters like this one `/cart?queryKey=queryValue` you build url like this `serviceUrl/queryKey=queryValue`

Reply:
I agree with you, need to fix it.